guardian/prosemirror-embeds
# Create node-based embeds with existing data

## What does this change?

At the moment, creating embeds with field content doesn't fill embeds with that content, you just get a blank embed.

This PR fixes that, creating the necessary nodes according to their specification.

There's not a way to do this in the testing UI for now, but I've added unit tests that should convincingly show that this is possible and works as expected, including typesafety for the added attributes. It can also be tested by changing code in `index.ts`.

## How to test

- The automated tests should pass. They should be comprehensive enough that you're happy this works.
- Run the test server with `yarn start`, and experiment with the default values provided in `index.ts` on L70-72. Does altering these values and inserting embeds match the expected behaviour? (Don't forget that rich text fields expect HTML input!)
